R/archivr.R

```diff
@@ -143,41 +124,39 @@ list_string <- function (url_list) {
   return (paste0("'[", string, "]'"))
 }
 
-#' Saves a single url in either perma.cc or the wayback machine.
+#' Saves a single url in perma.cc.
 #'
 #' @param arc_url The url to archive.
-#' @param method Either "perma_cc" or the default, "wayback."
 #' @importFrom jsonlite fromJSON
 #' @import curl
 #' @export
 #' @return A list or object representing the result.
-archiv_url <- function (arc_url, method="perma_cc") {
+archiv_perma <- function (arc_url, method="perma_cc") {
   api <- get_api_key()
   fold <- toString(get_folder_id())
-  if (method == "perma_cc") {
-    folder_url <- paste0()
-    api_url <- paste0(.perma_cc_post_api_url, api)
-    setting <- new_handle()
-    handle_setopt(setting, customrequest = "POST")
-    handle_setform(setting, url = arc_url, folder = fold)
-    result <- list(arc_url, "noguid", "unknown", "no url", "no screenshot", "no short url")
-    r <- curl_fetch_memory(api_url, setting)
-    reply <- fromJSON(rawToChar(r$content))
-    if ((!(is.null(reply$detail))) && reply$detail == "Authentication credentials were not provided.") {
-      print("Please input your api key:\nUse 'set_api_key(API_KEY)'")
-    } else if ((!(is.null(reply$error)))) {
-      print(reply)
-      print("Received an error reply, likely because your limit has been exceeded.")
-    } else {
-      if (!(is.null(reply$url == "Not a valid URL."))) {
-        result <- c(reply$url, reply$guid, reply$archive_timestamp,
-          reply$captures[1,]$playback_url, reply$captures[2,]$playback_url,
-        paste0("https://perma.cc/", reply$guid))
-      }
-      return(result)
+  if (is.null(api) || api == "") {
+    stop("API key not set for perma.cc. Use 'set_api_key() to set your key before using method='perma_cc'")
+  }
+  folder_url <- paste0()
+  api_url <- paste0(.perma_cc_post_api_url, api)
+  setting <- new_handle()
+  handle_setopt(setting, customrequest = "POST")
+  handle_setform(setting, url = arc_url, folder = fold)
+  result <- list(arc_url, "noguid", "unknown", "no url", "no screenshot", "no short url")
+  r <- curl_fetch_memory(api_url, setting)
+  reply <- fromJSON(rawToChar(r$content))
+  if ((!(is.null(reply$detail))) && reply$detail == "Authentication credentials were not provided.") {
+    stop("Please input your api key:\nUse 'set_api_key(API_KEY)'")
+  } else if ((!(is.null(reply$error)))) {
+    print(reply)
```

Comment:
Probably worth removing this print statement (my bad).

Reply:
No, that actually printed on an error I got from perma.cc and was quite helpful.

Reply:
Sounds good.